### PR TITLE
Redirect with modal open on form error.

### PR DESCRIPTION
**Pull request by DFurnes**
Redirects to page with modal hash in URL if there's a form error (will trigger modal on load with updates in Neue).

### CODE REVIEW

**Comment by barryclark** [comment]
:+1: !

**Comment by weerd** [comment]
:sun_with_face: :+1:

File: lib/modules/dosomething/dosomething_signup/dosomething_signup.signup_data_form.inc [hunk 1]
```diff
@@ -223,11 +223,13 @@ function dosomething_signup_data_validate_address($form, &$form_state) {
 
   // Did we not get any results?
   if (in_array('sorry', $formatted_address)) {
+    drupal_goto($_SERVER['REQUEST_URI'], array('fragment' => 'modal-signup-data-form'));
```
Comment (aaronschachter) [inline lib/modules/dosomething/dosomething_signup/dosomething_signup.signup_data_form.inc]: Won't this reload the page and the user has to fill out the form again?

Comment (DFurnes) [inline lib/modules/dosomething/dosomething_signup/dosomething_signup.signup_data_form.inc]: Yup, but only on form errors (which would currently reload the page but not include the modal hash at the end).

Comment (aaronschachter) [inline lib/modules/dosomething/dosomething_signup/dosomething_signup.signup_data_form.inc]: The difference for current behavior though is that the form reloads with the user input still filled out (but they do need to figure out to click on the link again).  Not going to keep me up at night or anything, its just something we lose by doing the `drupal_goto`.

**Comment by DFurnes** [comment]
So this isn't great, but it should be fine enough for now. If Drupal outputs a `.messages` class at the top of the page, we copy it into a container in the modal. We lose the user's filled form state

screen shot 2014-04-18 at 1 55 32 pm

**Comment by DFurnes** [comment]
Gonna merge this guy in so that @mikefantini can check it out on staging. This is only a temporary fix until we can run the address validation as a client-side check before submitting the form, so it's a bit messy but not permanent.

**Comment by barryclark** [comment]
Nice job @DFurnes @angaither :+1:

**Pull request merged**